storage/internal/benchmarks/upload_benchmark.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"cloud.google.com/go/storage"
+)
+
+type uploadParams struct {
+	o              *storage.ObjectHandle
+	contentsReader io.ReadSeeker
+	readerLength   int
+	objectSize     int64
+	chunkSize      int
+}
+
+func uploadBenchmark(ctx context.Context, u uploadParams) (time.Duration, error) {
+	elapsedTime := time.Duration(0)
+	start := time.Now()
+
+	u.o.If(storage.Conditions{
+		DoesNotExist: true,
```

Comment:
It is interesting that we end up including retried requests in the benchmark. I'm not sure what other languages do in that regard.

Reply:
Yep. We should probably discuss with the larger team so we implement it consistently.

Reply:
Added it as a topic for next team meeting.